Cassiopee/Generator/Generator/IBM.py

```diff
@@ -934,22 +940,28 @@ def addRefinementZones__(o, tb, tbox, snearsf, vmin, dim):
     volmin0 = 1.*volmin0
     while end == 0:
         # Do not refine inside obstacles
-        C._initVars(to, 'centers:cellN', 1.)
-        to = X_IBM.blankByIBCBodies(to, tbSolid, 'centers', dim)
+        C._initVars(to,'cellN',1.)
+        to = X_IBM.blankByIBCBodies(to, tbSolid, 'nodes', 3)
+        to = C.node2Center(to, 'cellN')
+        Internal._rmNodesFromName(to, Internal.__FlowSolutionNodes__)
+        C._initVars(to, '{centers:cellN}=({centers:cellN}>0.9)')
         C._initVars(to, '{centers:cellNBody}={centers:cellN}')
         nob = 0
         C._initVars(to, 'centers:indicator', 0.)
         for box in boxes:
             volmin2 = 1.09*(snearsf[nob])**(dim)
-            C._initVars(to,'centers:cellN',1.)
+            C._initVars(to,'cellN',1.)
             tboxl = C.newPyTree(['BOXLOC']); tboxl[2][1][2] = box
-            to = X_IBM.blankByIBCBodies(to, tboxl, 'centers', dim)
+            to = X_IBM.blankByIBCBodies(to, tboxl, 'nodes', 3)
+            to = C.node2Center(to, 'cellN')
+            C._initVars(to, '{centers:cellN}=({centers:cellN}>0.9)')
+            Internal._rmNodesFromName(to, Internal.__FlowSolutionNodes__)
             fact = 1.1
             while C.getMinValue(to, 'centers:cellN') == 1 and fact < 10.:
                 print("Info: addRefinementZones: tbox too small - increase tbox by fact = %2.1f"%(fact))
                 box2 = T.scale(box, fact) 
                 tboxl[2][1][2] = box2
-                to = X_IBM.blankByIBCBodies(to, tboxl, 'centers', dim)
+                to = X_IBM.blankByIBCBodies(to, tboxl, 'centers', 3)
```

Comment:
is this a typo or is 3 also used for 2D problems?

Reply:
It's not a typo, I use the node-centered blankCellsTri algorithm for both 2D and 3D cases! This also explains the addkplane at the beginning for 2D tboxes.